pkg/state/types.go

```diff
@@ -672,26 +673,59 @@ func (e EmptyInterfaceUsingUnderlyingType) Less(i, j int) bool {
 	return strings.Compare(objI, objJ) == -1
 }
 
-// Helper function to sort nested arrays in a map
-func sortNestedArrays(m map[string]interface{}) map[string]interface{} {
-	sortedMap := make(map[string]interface{})
+// Helper function to get schema for a field name
+func getSchemaForFieldName(schema gjson.Result, fieldName string) gjson.Result {
+	fieldsQuery := "fields.#(" + fieldName + ")." + fieldName
```

Comment:
Let's use constants for the strings that we don't want to be mutated like `"fields.#("`. Makes the code more readable too.